armory/datasets/preprocessing.py

```diff
@@ -39,22 +40,49 @@ def supervised_image_classification(element):
     return (image_to_canon(element["image"]), element["label"])
 
 
-mnist = register(supervised_image_classification, "mnist")
-cifar10 = register(supervised_image_classification, "cifar10")
-resisc45 = register(supervised_image_classification, "resisc45")
-
-
 @register
 def digit(element):
     return (audio_to_canon(element["audio"]), element["label"])
 
 
+@register
+def carla_obj_det_dev(element, modality="rgb"):
+    return carla_multimodal_obj_det(element["image"], modality=modality), (
+        convert_tf_obj_det_label_to_pytorch(element["image"], element["objects"]),
+        element["patch_metadata"],
+    )
+
+
 @register
 def carla_over_obj_det_dev(element, modality="rgb"):
-    return carla_over_obj_det_image(element["image"], modality=modality), (
+    x = carla_multimodal_obj_det(element["image"], modality=modality), (
         convert_tf_obj_det_label_to_pytorch(element["image"], element["objects"]),
         element["patch_metadata"],
     )
+    print(x)
+    return x
+
+
+@register
+def carla_video_tracking_dev(element, max_frames=None):
```

Comment:
I am assuming `max_frames` gets passed in as a `dataset_kwargs` but this is untested as I cannot find an example of it being used previously.